packages/next/cli/next-dev.ts

```diff
@@ -122,7 +126,9 @@ const nextDev: cliCommand = (argv) => {
     })
     .catch((err) => {
       if (err.code === 'EADDRINUSE') {
-        let errorMessage = `Port ${port} is already in use.`
+        let errorMessage = socket
+          ? `Socket path ${socket} already exists.`
```

Comment:
This is a common problem with listening on socket files. If the program doesn't close cleanly, it's very common to leave socket files around. There are 2 common ways to deal with this that are not mutually exclusive, afaik:
1. Use process init script to check/cleanup existing sockets. Systemd's `RuntimeDirectory=` option effectively does this by removing and recreating the directory on ever execution. By putting a socket file in there, you're guaranteed to never run into `EADDRINUSE`.
2. If you detect/catch this error, test the socket file for if it's still connectable with a client socket. If it's not connectable, the server that created the socket file is probably dead. So you can remove the socket file and try again. If it is connectable, close the connection and throw the original `EADDRINUSE`.

Reply:
I rebased and implemented the check.